### PR TITLE
Add PersistentTaskExtension interface

### PR DESCRIPTION
**Pull request by Kuanysh-kst**
### Description
_Describe what this change achieves._

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dbwiddis — APPROVED** [review]
Thanks for this contribution! LGTM!

**Comment by Kuanysh-kst** [comment]
it's normal that Gradle Build/check (pull_request) failed here

**Comment by dbwiddis** [comment]
> is it normal that Gradle Build/check (pull_request) failed here

We just added a new way of javadoc checks in #551 and a choice I made when submitting that was to require `@param` entries for extension points with parameters.#543 

I'm not sure this was the best choice, given that we are copying existing docs from the plugin interfaces, but for now the best thing to do is to update the javadoc with `@param` tags for methods which take parameters.

**Review by saratvemulapalli — APPROVED** [review]
@Kuanysh-kst thanks for the contribution. 
Looks like there are conflicts with main, could you rebase?

**Review by dbwiddis — APPROVED** [review]

File: src/main/java/org/opensearch/sdk/PersistentTaskExtension.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+/**
+ * Extension for registering persistent tasks executors.
+ */
+
+public interface PersistentTaskExtension {
+
+    /**
+     * Returns additional persistent tasks executors added by this extension.
+     */
+    default List<PersistentTasksExecutor<?>> getPersistentTasksExecutor(
```
Comment (dbwiddis) [inline src/main/java/org/opensearch/sdk/PersistentTaskExtension.java:31]: We need to add `@param` javadocs for these parameters.

**Review by dbwiddis — CHANGES_REQUESTED** [review]
Build failing on missing params in javadocs.

**Review by dbwiddis — APPROVED** [review]
LGTM! Thanks!

File: src/main/java/org/opensearch/sdk/PersistentTaskExtension.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+     * Returns additional persistent tasks executors added by this extension.
+     * @param clusterService the cluster service used to coordinate actions across the cluster.
+     * @param threadPool the thread pool used to execute tasks.
+     * @param client the client used to interact with the Elasticsearch cluster.
+     * @param settingsModule the module containing Elasticsearch settings.
+     * @param expressionResolver the resolver used to parse index name expressions.
+     * @return an empty List.
+     */
+    default List<PersistentTasksExecutor<?>> getPersistentTasksExecutor(
+        ClusterService clusterService,
```
[the remark this answers is not in the document]
Comment (dbwiddis) [reply src/main/java/org/opensearch/sdk/PersistentTaskExtension.java:38]: Eventually, when this extension point is implemented, yes.  No need to change it now, though.

**Review by owaiskazi19 — APPROVED** [review]

**Pull request merged**